diff -urN qtlocation-everywhere-src-5.15.9/src/3rdparty/mapbox-gl-native/.git qtlocation/src/3rdparty/mapbox-gl-native/.git
--- qtlocation-everywhere-src-5.15.9/src/3rdparty/mapbox-gl-native/.git	1970-01-01 01:00:00.000000000 +0100
+++ qtlocation/src/3rdparty/mapbox-gl-native/.git	2023-04-10 21:14:44.341376295 +0200
@@ -0,0 +1 @@
+gitdir: ../../../.git/modules/src/3rdparty/mapbox-gl-native
diff -urN qtlocation-everywhere-src-5.15.9/src/3rdparty/mapbox-gl-native/include/mbgl/util/geometry.hpp qtlocation/src/3rdparty/mapbox-gl-native/include/mbgl/util/geometry.hpp
--- qtlocation-everywhere-src-5.15.9/src/3rdparty/mapbox-gl-native/include/mbgl/util/geometry.hpp	2023-03-23 15:00:13.000000000 +0100
+++ qtlocation/src/3rdparty/mapbox-gl-native/include/mbgl/util/geometry.hpp	2023-04-10 21:14:44.499377600 +0200
@@ -1,5 +1,7 @@
 #pragma once
 
+#include <cstdint>
+
 #include <mapbox/geometry/geometry.hpp>
 #include <mapbox/geometry/point_arithmetic.hpp>
 #include <mapbox/geometry/for_each_point.hpp>
diff -urN qtlocation-everywhere-src-5.15.9/src/3rdparty/mapbox-gl-native/include/mbgl/util/string.hpp qtlocation/src/3rdparty/mapbox-gl-native/include/mbgl/util/string.hpp
--- qtlocation-everywhere-src-5.15.9/src/3rdparty/mapbox-gl-native/include/mbgl/util/string.hpp	2023-03-23 15:00:13.000000000 +0100
+++ qtlocation/src/3rdparty/mapbox-gl-native/include/mbgl/util/string.hpp	2023-04-10 21:14:44.500377608 +0200
@@ -3,6 +3,7 @@
 #include <sstream>
 #include <string>
 #include <cassert>
+#include <cstdint>
 #include <cstdlib>
 #include <exception>
 
diff -urN qtlocation-everywhere-src-5.15.9/src/3rdparty/mapbox-gl-native/include/mbgl/util/unique_any.hpp qtlocation/src/3rdparty/mapbox-gl-native/include/mbgl/util/unique_any.hpp
--- qtlocation-everywhere-src-5.15.9/src/3rdparty/mapbox-gl-native/include/mbgl/util/unique_any.hpp	2023-03-23 15:00:13.000000000 +0100
+++ qtlocation/src/3rdparty/mapbox-gl-native/include/mbgl/util/unique_any.hpp	2023-04-21 19:29:34.677755219 +0200
@@ -3,6 +3,7 @@
 #include <typeinfo>
 #include <type_traits>
 #include <stdexcept>
+#include <utility>
 namespace mbgl {
 namespace util {
 
diff -urN qtlocation-everywhere-src-5.15.9/src/3rdparty/mapbox-gl-native/src/mbgl/gl/stencil_mode.hpp qtlocation/src/3rdparty/mapbox-gl-native/src/mbgl/gl/stencil_mode.hpp
--- qtlocation-everywhere-src-5.15.9/src/3rdparty/mapbox-gl-native/src/mbgl/gl/stencil_mode.hpp	2023-03-23 15:00:13.000000000 +0100
+++ qtlocation/src/3rdparty/mapbox-gl-native/src/mbgl/gl/stencil_mode.hpp	2023-04-10 21:14:44.506377657 +0200
@@ -1,5 +1,7 @@
 #pragma once
 
+#include <cstdint>
+
 #include <mbgl/util/variant.hpp>
 
 namespace mbgl {
